cms/templatetags/cms_tags.py

```diff
@@ -96,9 +95,10 @@ def _get_page_by_untyped_arg(page_lookup, request, site_id):
         if settings.DEBUG:
             raise Page.DoesNotExist(body)
         else:
+            mw = getattr(settings, 'MIDDLEWARE', getattr(settings, 'MIDDLEWARE_CLASSES'))
```

Comment:
can you add a utility to `cms.utils.compat.dj` that returns the middleware classes?